tensorflow/compiler/tf2tensorrt/utils/trt_tensor_proxy.h

```diff
@@ -0,0 +1,599 @@
+  FakeITensor* fake_tensor() const
+  {
+    assert(fake_tensor_ != nullptr);
+    assert(ttype_ == TensorType::kTRT);
+    return fake_tensor_.get();
+  }
+  
+  void setName(const char* name)
+  {
+    switch(ttype_)
+    {
+      case TensorType::kTRT: return trt_tensor_->setName(name);
+      case TensorType::kSIMPLE: return simple_tensor_->setName(name);
+      case TensorType::kFAKE: return fake_tensor_->setName(name);
+      case TensorType::kNONE: assert(0 && "Unsupported itensor_ type");
+    }
+    assert(0 && "Should not come here.");
+  }
+
+  const char* getName() const
+  {
+    switch(ttype_)
+    {
+      case TensorType::kTRT: return trt_tensor_->getName();
+      case TensorType::kSIMPLE: return simple_tensor_->getName();
+      case TensorType::kFAKE: return fake_tensor_->getName();
+      case TensorType::kNONE: assert(0 && "Unsupported itensor_ type");
+    }
+    assert(0 && "Should not come here.");
+  }
+
+  void setDimensions(nvinfer1::Dims dimensions)
+  {
+    switch(ttype_)
+    {
+      case TensorType::kTRT: return trt_tensor_->setDimensions(dimensions);
+      case TensorType::kSIMPLE: return simple_tensor_->setDimensions(dimensions);
+      case TensorType::kFAKE: return fake_tensor_->setDimensions(dimensions);
+      case TensorType::kNONE: assert(0 && "Unsupported itensor_ type");
+    }
+    assert(0 && "Should not come here.");
+  }
+
+  nvinfer1::Dims getDimensions() const
+  {
+    switch(ttype_)
+    {
+      case TensorType::kTRT: return trt_tensor_->getDimensions();
+      case TensorType::kSIMPLE: return simple_tensor_->getDimensions();
+      case TensorType::kFAKE: return fake_tensor_->getDimensions();
+      case TensorType::kNONE: assert(0 && "Unsupported itensor_ type");
+    }
+    assert(0 && "Should not come here.");
+  }
+
+  void setType(nvinfer1::DataType type)
+  {
+    switch(ttype_)
+    {
+      case TensorType::kTRT: return trt_tensor_->setType(type);
+      case TensorType::kSIMPLE: return simple_tensor_->setType(type);
+      case TensorType::kFAKE: return fake_tensor_->setType(type);
+      case TensorType::kNONE: assert(0 && "Unsupported itensor_ type");
+    }
+    assert(0 && "Should not come here.");
+  }
+
+  nvinfer1::DataType getType() const
+  {
+    switch(ttype_)
+    {
+      case TensorType::kTRT: return trt_tensor_->getType();
+      case TensorType::kSIMPLE: return simple_tensor_->getType();
+      case TensorType::kFAKE: return fake_tensor_->getType();
+      case TensorType::kNONE: assert(0 && "Unsupported itensor_ type");
+    }
+    assert(0 && "Should not come here.");
+  }
+
+  bool isNetworkInput() const
+  {
+    switch(ttype_)
+    {
+      case TensorType::kTRT: return trt_tensor_->isNetworkInput();
+      case TensorType::kSIMPLE: return simple_tensor_->isNetworkInput();
+      case TensorType::kFAKE: return fake_tensor_->isNetworkInput();
+      case TensorType::kNONE: assert(0 && "Unsupported itensor_ type");
+    }
+    assert(0 && "Should not come here.");
```

Comment:
This routine can't be compiled, you need to return a value at the end as well. 
A much better way is to use exhaustive switch, see here https://abseil.io/tips/147
for the case here, you can add a "return" on the kNONE branch. 
Previously, I also suggest to remove kNONE from the enum as there is no such a need. I also suggest to remove kFAKE (only used by tests)and make kSIMPLE work for the tests.

Similar problems in many other routine in this file.

Reply:
Thanks for the suggestion, after implementing this, removing the kNONE type, and merging the simple and fake tensors, these checks look much more streamlined